### PR TITLE
BUG: groupby not accepting list[str] as by argument

### LINKED ISSUE #187: BUG: groupby by argument raises with mypy when argument is annotated as ``list[str]``
**Describe the bug**

Using ``df.groupby(by=cols)`` where cols is annotated as ``list[str]`` raises a mypy error

```
error: Argument "by" to "groupby" of "DataFrame" has incompatible type "List[str]"; expected "Optional[Union[Tuple[Any, ...], List[Optional[Hashable]], Union[ufunc, Callable[..., Any]], Series[Any], ndarray[Any, Any], Mapping[Optional[Hashable], Any], Index]]"  [arg-type]
``` 

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
2. Indicate which type checker you are using (`mypy` or  `pyright`).
3. Show the error message received from that type checker while checking your example.

```
def _call_groupby(df: pd.DataFrame, cols: list[str]):
    df.groupby(by=cols).count()
```

This also raises when adding:

```
    cols: list[str] = ["col1", "col2"]
    df.groupby(by=cols).sum()
```

to ``test_types_groupby`` in ``test_frame.py``

``list[str]`` should be valid? If passing simply ``["col1", "col2"]`` without an annotation this works

**Please complete the following information:**
 - OS:  ubuntu
 - OS Version 22.04
 - python version 3.10
 - version of type checker mypy 0.971
 - version of installed `pandas-stubs` newest 20220807

**Additional context**

### PR DESCRIPTION
**Pull request by phofl**
- [x] Closes #187 (Replace xxxx with the Github issue number)
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

Please let me know, if there is anything else I should add

cc @twoertwein

### CODE REVIEW

File: tests/test_frame.py [hunk 1]
```diff
@@ -546,6 +546,13 @@ def test_types_groupby() -> None:
     df.groupby(by="col1")
     df.groupby(level="ind")
     df.groupby(by="col1", sort=False, as_index=True)
+
+    cols: list[str] = ["col1", "col2"]
+    df.groupby(by=cols).sum()
+
+    cols_opt: list[str | None] = ["col1", "col2"]
+    df.groupby(by=cols_opt).sum()
```
Comment (twoertwein) [inline tests/test_frame.py:554]: If you want, you can also test the return value

**Review by twoertwein — APPROVED** [review]

File: tests/test_frame.py (hunk 1 above)
Comment (phofl) [reply tests/test_frame.py]: Added

Comment (twoertwein) [reply tests/test_frame.py]: `check(assert_type(df.groupby(by=cols).sum(), pd.DataFrame), pd.DataFrame)`

Comment (phofl) [reply tests/test_frame.py]: Wouldn't this also fail with the assignment? Is the assert_type way the proper way of checking this?

Comment (twoertwein) [reply tests/test_frame.py]: If the return type was `Any`, `df: pd.DataFrame` would not report an error.

Comment (phofl) [reply tests/test_frame.py]: Ah, did not think about that. Thanks

Comment (phofl) [reply tests/test_frame.py]: Independently of this pr: We should check everything like this then, correct?

Comment (twoertwein) [reply tests/test_frame.py]: > Independently of this pr: We should check everything like this then, correct?

Ideally yes, but as a starting point, we like new code to use `assert_type` (for static typing) and also `check` (to test the runtime behavior; and for sequence-like items to check the dtpye of the items).

File: tests/test_frame.py [hunk 2]
```diff
@@ -565,6 +565,11 @@ def test_types_groupby() -> None:
     df10: pd.DataFrame = df.groupby("col1").agg(
         new_col=pd.NamedAgg(column="col2", aggfunc="max")
     )
+    cols: list[str] = ["col1", "col2"]
```
Comment (Dr-Irv) [inline tests/test_frame.py:568]: can you add a #GH 187 comment here?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
One small change, otherwise looks good

File: tests/test_frame.py (hunk 2 above)
Comment (phofl) [reply tests/test_frame.py:569]: Done

**Review by Dr-Irv — APPROVED** [review]
Thanks @phofl

**Pull request merged**